src/components/Filter/filter.jsx

```diff
@@ -0,0 +1,180 @@
+import { useFilter } from "../../context/filtercontext";
+
+export const Filter = () => {
+  const { state, dispatch } = useFilter();
+  const { price, rating, sortBy, categories } = state;
+  const { running, tennis, workout } = categories;
+
+  return (
+    <sidebar>
+      <div className="sidebar">
+        <div className="filter-header flex-row filter-styling">
+          <p className="h3">Filters</p>
+          <button onClick={() => dispatch({ type: "CLEAR" })}>
+            <p className="h4">Clear All Filters</p>
+          </button>
+        </div>
+
+        <div className="filter-price filter-styling">
+          <p className="h3">Price</p>
+          <div className="filter-price-break flex-row">
+            <p className="h3">1k</p>
+            <p className="h3">2k</p>
+            <p className="h3">3k</p>
+            <p className="h3">4k</p>
+            <p className="h3">5k</p>
+          </div>
+
+          <div className="slidecontainer">
+            <input
+              type="range"
+              min="1000"
+              max="5000"
+              defaultValue="50"
+              step="1000"
+              value={price}
+              className="slider"
+              id="myRange"
+              onChange={(e) =>
+                dispatch({ type: "PRICE", price_val: e.target.value })
+              }
+            />
+          </div>
+        </div>
+
+        <div className="filter-category flex-col filter-styling">
+          <p className="h3">Category</p>
+          <div className="filter-checkboks checkbox">
+            <label for="checkbox-1">
+              <input
+                id="checkbox-1"
+                name="running"
+                type="checkbox"
+                value="running"
+                checked={running}
+                onChange={(e) => dispatch({ type: "RUNNING" })}
+              />
+              Running
+            </label>
+            <label for="checkbox-1">
+              <input
+                id="checkbox-1"
+                name="workout"
+                type="checkbox"
+                value="workout"
+                checked={workout}
+                onChange={(e) => dispatch({ type: "WORKOUT" })}
+              />
+              Workout
+            </label>
+            <label for="checkbox-1">
+              <input
+                id="checkbox-1"
+                name="tennis"
+                type="checkbox"
+                value="tennis"
+                checked={tennis}
+                onChange={(e) => dispatch({ type: "TENNIS" })}
+              />
```

Comment:
Here you have provided same id and for attribute for all of three categories. Because of that when i clicked on any one of last two categories, it is selecting only the first category. Please check this